include/uitsl/debug/EnumeratedFunctor.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <unordered_map>
+
+#include "../third-party/Empirical/source/base/vector.h"
+
+namespace uitsl {
+
+template <typename Input, typename Output>
+class EnumeratedFunctor {
```

Comment:
this will probably be a confusing class in a few months XD can you add a one sentence summary for the class?